_posts/2017-01-21-Wonders-of-Static-Linking.md

```diff
@@ -0,0 +1,147 @@
+
+So lets make an illustration, a program to make the static library. This very simple program will teach you the basics. So ppack youself.
+
+Consider we want to make a very simple library the math library, not so hifi, very low level ;-), which just has 4 operations add, sub, multi and div.
+
+So for this we will need a name for library, let it be poseidonMath, lets make poseidonMath.h first
+
+{% highlight c %}
+int add(int, int);
+int sub(int, int);
+int multi(int, int);
+int div(int, int);
+{% endhighlight %}
+
+and this is the header file of the poseidonMath library, now we make the files for the functions we make add.c, sub.c, multi.c and div.c
+
+so here are
+
+add.c
+
+{% highlight c %}
+int add(int a, int b)
+{
+    return a + b;
+}
+{% endhighlight %}
+
+sub.c
+
+{% highlight c %}
+int sub(int a, int b)
+{
+    return a - b;
+}
+{% endhighlight %}
+
+multi.c
+
+{% highlight c %}
+int multi(int a, int b)
+{
+    return a * b;
+}
+{% endhighlight %}
+
+div.c
+
+{% highlight c %}
+int div(int a, int b)
+{
+    return a / b;
+}
+{% endhighlight %}
+
+Now we have the files we can now the compile them, **JUST COMPILE** them, just compiling will make the .o or .obj files, these will have just the assembly code of the files, these will not be able to run as they don't have the main, they are just the piece of the code in assembly, nothing else, just the parts of aircraft made separately in different industries, nothing else, so how to compile them, just do the following, I hope in linux this following command is self explainary, -c tells just compile.
+
+{% highlight bash %}
+
+$ gcc -c add.c
+$ gcc -c sub.c
+$ gcc -c multi.c
+$ gcc -c div.c
+{% endhighlight %}
+
+Now, they will output the add.o, sub.o, multi.o and div.o files, the object files like parts(object codes) made from the blueprints(source codes). Its time to make the file for the main function, so here it is,
+
+main.c
+
+{% highlight c %}
+
+#include <poseidonMath.h>
+#include <stdio.h>
+int main()
+{
+    int a = 10, b = 2;
+    printf("%d %d %d %d\n", add(a, b), sub(a, b), multi(a, b), div(a, b));
+    return 0;
+}
+{% endhighlight %}
+
+The output will be clear for just begineer, for clearity it will `12 8 20 5` , but wait I tell this will be the output, sorry I was wrong, we first have to compile it, now how to do it.
```

Comment:
> begineer

beginner

> clearity

clarity

> but wait I tell this will be the output, sorry I was wrong, we first have to compile it, now how to do it.

I don't understand what you're trying to say here.